src/jsn.erl

```diff
@@ -920,12 +920,19 @@ key_get(_Key, _List, Default) ->
 %%------------------------------------------------------------------------------
 get_nth(_Index, [], Default) ->
     Default;
-get_nth(first, [H|_], _Default) when ?IS_SIMPLE_JSON_TERM(H) ->
-    H;
-get_nth(last, [H|_] = A, _Default) when ?IS_SIMPLE_JSON_TERM(H) ->
-    hd(lists:reverse(A));
-get_nth(Index, A, _Default) when Index > 0, Index =< length(A) ->
-    lists:nth(Index, A);
+get_nth(first, A, Default) ->
+    get_nth(1, A, Default);
+get_nth(last, A, Default) when is_list(A) ->
+    get_nth(1, lists:reverse(A), Default);
```

Comment:
Maybe consider `lists:last` here, as it should be more efficient than reversing the entire list for a single element.

Reply:
@mrallen1 Good point. I pulled out the logic of validating that the array is a potentially JSON array element into a separate function (primarily as opposed to a proplist element), `get_if_valid`. I'm mulling over whether that's worth just also adding to `key_get/3` above it.

Anyway, that lets `get_nth/3` reuse that logic in all the heads, I believe as efficiently as possible. Cheers.